nucypher/blockchain/eth/agents.py

```diff
@@ -321,15 +330,36 @@ def get_all_active_stakers(self, periods: int, pagination_size: Optional[int] =
             start_index: int = 0
             n_tokens: int = 0
             stakers: Dict[int, int] = dict()
-            active_stakers: Tuple[NuNits, List[List[int]]]
+            attempts = 0
```

Comment:
Should this be set inside the `while` loop? Or do we only get three pagination decreases in total?

Reply:
Three pagination decreases in total. Because of the start_index, if a larger size worked once but not a subsequent time, we just continue from where we were with a decreased size e.g. if 250 pagination size worked the first time (i.e. we got the first 250 stakers), and then failed the second time for a timeout reason, we would try again with a pagination size of 125 starting from start_index 250.